From 1a16da4ed0913c2de36b070326277a3aa555e376 Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Thu, 13 Jul 2023 16:35:40 -0300
Subject: [PATCH] Add ServiceAccount to openvswitch pod

This change includes back the the helm-toolkit snippet:
kubernetes_pod_rbac_serviceaccount to the openvswitch Daemonset
definition, since it is responsible for creating the POD's
ServiceAccount which contains imagePullSecrets that enable the POD to
retrieve images from private registries.

Originally openvswitch chart had two daemonset definitions: for the
db and for the server, but recently both were merged into a single
daemonset [1] and the template inclusion was dropped during this merge

[1] https://github.com/openstack/openstack-helm-infra/
    commit/73e2b3322d3cc8ca4ee8453dd612266589d734b1

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Change-Id: I59ef955529a34020d4bba0da0b48d1f47e2e1039
---
 openvswitch/templates/daemonset.yaml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/openvswitch/templates/daemonset.yaml b/openvswitch/templates/daemonset.yaml
index 244ffb8e..bd37de14 100644
--- a/openvswitch/templates/daemonset.yaml
+++ b/openvswitch/templates/daemonset.yaml
@@ -56,6 +56,9 @@ exec:
 
 {{- if .Values.manifests.daemonset }}
 {{- $envAll := . }}
+
+{{- $serviceAccountName := "openvswitch-server" }}
+{{ tuple $envAll "vswitchd" $serviceAccountName | include "helm-toolkit.snippets.kubernetes_pod_rbac_serviceaccount" }}
 ---
 apiVersion: apps/v1
 kind: DaemonSet
@@ -257,4 +260,4 @@ It should be handled through lcore and pmd core masks. */}}
           hostPath:
             path: /sys/fs/cgroup
 {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
-- 
2.25.1

